2025-07-15 00:04:07,500 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:04:17,889 : ERROR : Exception on /login [GET] [in C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py : 875]
Traceback (most recent call last):
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\routes.py", line 56, in login
    return render_template('login.html', form= _form)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\login.html", line 2, in top-level template code
    {% import bootstrap_wtf.html as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'bootstrap_wtf' is undefined
2025-07-15 00:06:06,932 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:13:38,593 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:18:22,068 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:18:26,550 : ERROR : Exception on /index [GET] [in C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py : 875]
Traceback (most recent call last):
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\routes.py", line 31, in index
    return render_template('index.html', title='home page', posts= posts.items, form = postForm, prev_url= prev_url, next_url= next_url)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 149, in render_template
    template = app.jinja_env.get_or_select_template(template_name_or_list)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 1087, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 1016, in get_template
    return self._load_template(name, globals)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 975, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\loaders.py", line 138, in load
    code = environment.compile(source, name, filename)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 771, in compile
    self.handle_exception(source=source_hint)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\index.html", line 13, in template
    {% include '_posts.html' %}
    ^^^^^^^^^^^^^^
jinja2.exceptions.TemplateSyntaxError: expected token ',', got '_posts'
2025-07-15 00:27:21,885 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
2025-07-15 00:27:26,568 : ERROR : Exception on /index [GET] [in C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py : 875]
Traceback (most recent call last):
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\routes.py", line 31, in index
    return render_template('index.html', title='home page', posts= posts.items, form = postForm, prev_url= prev_url, next_url= next_url)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Kakashi\Desktop\projects\newblog\app\templates\index.html", line 2, in top-level template code
    {% import 'bootstrap_wtf' as wtf %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 65, in get_source
    return self._get_source_fast(environment, template)
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Kakashi\Desktop\projects\newblog\venv\Lib\site-packages\flask\templating.py", line 99, in _get_source_fast
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: bootstrap_wtf
2025-07-15 00:28:05,806 : INFO : Microblog StartUP [in C:\Users\Kakashi\Desktop\projects\newblog\app\__init__.py : 30]
